sphinx/writers/latex.py

```diff
@@ -1500,9 +1517,13 @@ def get_nested_level(node):
             else:
                 return get_nested_level(node.parent)
 
+        enum_depth = "enum%s" % toRoman(get_nested_level(node)).lower()
+        enum_label = ENUMERATE_LABELS.get(get_enumtype(node))
+
         self.body.append('\\begin{enumerate}\n')
+        self.body.append('\\renewcommand{\\label%s}{%s%s{%s}%s}\n' %
+                         (enum_depth, node['prefix'], enum_label, enum_depth, node['suffix']))
```

Comment:
Some discussion is needed here because this will break the `\label/\ref` system. However out-of-the hat it is possible that Sphinx uses mainly `\hyperref` for hyperlinks, it doesn't rely on LaTeX's `\ref` being made hyperlink by package hyperref (I have volatile memory and should review latex.py first). Here is extract from LaTeX source code:

```
Enumeration is done with four counters: enumi, enumii, enumiii and enumiv,
where enumN controls the numbering of the Nth level enumeration. The label
is generated by the commands \labelenumi . . . \labelenumiv, which should be
defined by the document style. Note that \p@enumN\theenumN defines the output
of a \ref command. A typical definition might be:

\def\theenumii{\alph{enumii}}
\def\p@enumii{\theenumi}
\def\labelenumii{(\theenumii)}

which will print the labels as `(a)', `(b)', . . . and print a \ref as `3a'.
```

Thus it would be advisable to modify rather `\theenumi` to be what one wants (without the dot). If one leaves `\labelenumii` untouched for example then at second depth level `A. ` in source will become `(A)` in output (at least in English language and no additional package). WIth present PR, it would become `A. ` rather. And references from usage of `\ref` elsewhere in document would be wrong if a `\label` is located there. (but again, I am not sure writers/latex.py uses anything of `\label/\ref`).

Reply:
Another remark is that with present PR input such as this

```
a. first

b. second

c. third

   A. subfirst
   B. subsecond
   C. subthird

      #. subsub
      #. subsub
      #. subsub
```

will give in output at third level `1.`, `2.`,  `3.` whereas LaTeX's default (for English language at least) would be at third level to do `i.`, `ii.`, `iii.`.

Reply:
About my previous comment, output in PDF in `1.`, `2.`, `3.` at 3rd level from `#.` in source is same as in HTML, so arguably this is +1.